### PR TITLE
fix: ES Rest Client Creation for non ssl authenticated connection

### PR DESCRIPTION
**Pull request by nj7**
Fixed bug stated in https://github.com/datahub-project/datahub/issues/5046

### CODE REVIEW

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java [hunk 1]
```diff
@@ -89,6 +91,35 @@ private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int po
...
+      int connectionRequestTimeout, String username, String password) {
+    RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "http"))
+        .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder
+            .setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(threadCount).build()));
+
+    if (!StringUtils.isEmpty(pathPrefix)) {
+      builder.setPathPrefix(pathPrefix);
+    }
+    
+    builder.setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
```
Comment (dexter-mh-lee) [inline metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java:106]: Seems like this is the only thing being added to the original one? Can we call the original function and then set the client config callback of the returned RestClientBuilder?

Comment (nj7) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java:106]: yup make, sense

**Review by dexter-mh-lee — APPROVED** [review]
LGTM! Thanks for the contribution!!!

File: metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java [hunk 2]
```diff
@@ -89,6 +91,26 @@ private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int po
 
     return builder;
   }
+  
+  @Nonnull
+  private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int port, String pathPrefix, int threadCount,
+      int connectionRequestTimeout, String username, String password) {
+    builder = loadRestHttpClient(host, port, pathPrefix, threadCount, connectionRequestTimeout);
```
[the remark this answers is not in the document]
Comment (nj7) [reply metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java:98]: working on it

**Pull request merged**